ext/dom/node.c

```diff
@@ -876,29 +952,15 @@ static xmlNodePtr _php_dom_insert_fragment(xmlNodePtr nodep, xmlNodePtr prevsib,
 /* {{{ URL: http://www.w3.org/TR/2003/WD-DOM-Level-3-Core-20030226/DOM3-Core.html#core-ID-952280727
 Since:
 */
-PHP_METHOD(DOMNode, insertBefore)
+static void dom_node_insert_before_legacy(zval *return_value, zval *ref, dom_object *intern, dom_object *childobj, xmlNodePtr parentp, xmlNodePtr child)
 {
-	zval *id, *node, *ref = NULL;
-	xmlNodePtr child, new_child, parentp, refp;
-	dom_object *intern, *childobj, *refpobj;
-	int ret, stricterror;
-
-	id = ZEND_THIS;
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|O!", &node, dom_node_class_entry, &ref, dom_node_class_entry) == FAILURE) {
-		RETURN_THROWS();
-	}
-
-	DOM_GET_OBJ(parentp, id, xmlNodePtr, intern);
-
-	if (dom_node_children_valid(parentp) == FAILURE) {
+	if (!dom_node_children_valid(parentp)) {
 		RETURN_FALSE;
 	}
 
-	DOM_GET_OBJ(child, node, xmlNodePtr, childobj);
-
-	new_child = NULL;
-
-	stricterror = dom_get_strict_error(intern->document);
+	xmlNodePtr new_child = NULL;
+	int stricterror = dom_get_strict_error(intern->document);
```

Comment:
Nit: 
```suggestion
	bool stricterror = dom_get_strict_error(intern->document);
```
Or is this old DOM code that has been copy pasted?